packages/component-base/src/virtualizer-iron-list-adapter.js

```diff
@@ -143,6 +143,35 @@ export class IronListAdapter {
     });
   }
 
+  _updateMetrics(itemSet) {
+    // Make sure we distributed all the physical items
+    // so we can measure them.
+    flush();
+
+    let newPhysicalSize = 0;
+    let oldPhysicalSize = 0;
+    const prevAvgCount = this._physicalAverageCount;
+    const prevPhysicalAvg = this._physicalAverage;
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    this._iterateItems((pidx, vidx) => {
+      oldPhysicalSize += this._physicalSizes[pidx];
+      const physicalItem = this._physicalItems[pidx];
+      this._physicalSizes[pidx] = Math.max(physicalItem.offsetHeight, physicalItem.getBoundingClientRect().height);
```

Comment:
Using `Math.max(physicalItem.offsetHeight, physicalItem.getBoundingClientRect().height)` here feels a bit arbitrary because it resolves the sub-pixel value as...
- the fractional value as such when < X.5
- the upwards rounded `offsetHeight` when > X.5

For example, an effective element height of `100.25` would be used as such, but `100.75` would get rounded up to `101`.

Because of this, the associated test case `should take sub-pixel value when measuring items height` wouldn't pass if you change the used element height from `30.25px` to `30.75px`.

If it fixes the issue, I'd suggest using `Math.ceil(physicalItem.getBoundingClientRect().height)` instead for consistency. Or alternatively, could figure out why some tests fail on headless Chrome when the value is used as such (without upwards rounding)?

Reply:
> The offsetHeight is still being used, because I noticed that sometimes, on the first paint, the value returned from physicalItem.getBoundingClientRect().height is way smaller. I wasn't able to create a test case for this scenario.

This could be caused by an overlay open animation or such. During the animation there may be some CSS transform applied to the overlay. `offsetHeight` isn't affected by transforms, but boundingClientRect is.

Maybe a test case could be added where the virtualizer has `transform: scale(0.5)`. I'd imagine using boundingClientRect only would end up having the items misplaced / wrongly measured.

Reply:
The new test uses `Math.round` while the other test uses `Math.ceil` which implies there's still something not right here.

One issue is that when scaled down, we always fall back to using `offsetHeight`, which could possibly again lead to the original "twerking" behavior.

Another issue is that if we scale up (`transform: scale(1.5)`), it would always use the bounding client rect height, which would be incorrect and lead to mispositioned items (scaling up also needs a test case!).

In practice, `getBoundingClientRect()` is only viable when there's no scaling involved, so we can't really use it after all.

Let's instead use computed style, which is a bit more verbose, but it should give us the height in sub-pixel precision and it's not affected by transform:

```js
const style = getComputedStyle(physicalItem);
let totalHeight = parseFloat(style.height) || 0;
if (style.boxSizing !== 'border-box') {
  totalHeight +=
    parseFloat(style.paddingTop) +
    parseFloat(style.paddingBottom) +
    parseFloat(style.borderTopWidth) +
    parseFloat(style.borderBottomWidth);
}
this._physicalSizes[pidx] = Math.ceil(totalHeight);
```

This needs some additional tests to cover the custom border-box height calculation when "box-sizing: border-box" isn't used.

Reply:
Done.

I was actually starting to consider this approach because of the issues you described. Now it should be more reliable.